src/services/requestSessions.js

```diff
@@ -0,0 +1,564 @@
+			})
+		} catch (error) {
+			console.error(error)
+			throw error
+		}
+	}
+
+	/**
+	 * Get information about a session between the authenticated user and another user.
+	 * @param {Object} req - The request object.
+	 * @param {Object} req.body - The body of the request.
+	 * @param {string} req.body.user_id - The ID of the user to get connection info for.
+	 * @param {Object} req.decodedToken - The decoded token containing authenticated user info.
+	 * @param {string} req.decodedToken.id - The ID of the authenticated user.
+	 * @returns {Promise<Object>} The session information.
+	 * @throws Will throw an error if the request fails.
+	 */
+	static async getInfo(friendId, userId, startDate, endDate, status) {
+		try {
+			const requestSessions = await sessionRequestQueries.getRequestSessions(
+				userId,
+				friendId,
+				startDate,
+				endDate,
+				status
+			)
+
+			const defaultOrgId = await getDefaultOrgId()
+			if (!defaultOrgId) {
+				return responses.failureResponse({
+					message: 'DEFAULT_ORG_ID_NOT_SET',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			const [userExtensionsModelName, userDetails] = await Promise.all([
+				userExtensionQueries.getModelName(),
+				userExtensionQueries.getMenteeExtension(friendId, [
+					'name',
+					'user_id',
+					'mentee_visibility',
+					'organization_id',
+					'designation',
+					'area_of_expertise',
+					'education_qualification',
+					'custom_entity_text',
+					'meta',
+					'is_mentor',
+					'experience',
+					'image',
+				]),
+			])
+
+			if (requestSessions?.status === common.CONNECTIONS_STATUS.BLOCKED || !userDetails) {
+				return responses.successResponse({
+					statusCode: httpStatusCode.ok,
+					message: 'USER_NOT_FOUND',
+				})
+			}
+			userDetails.image &&= (await userRequests.getDownloadableUrl(userDetails.image))?.result
+
+			// Fetch entity types associated with the user
+			let entityTypes = await entityTypeQueries.findUserEntityTypesAndEntities({
+				status: 'ACTIVE',
+				organization_id: {
+					[Op.in]: [userDetails.organization_id, defaultOrgId],
+				},
+				model_names: { [Op.contains]: [userExtensionsModelName] },
+			})
+			const validationData = removeDefaultOrgEntityTypes(entityTypes, userDetails.organization_id)
+			const processedUserDetails = utils.processDbResponse(userDetails, validationData)
+
+			if (!requestSessions) {
+				return responses.successResponse({
+					statusCode: httpStatusCode.ok,
+					message: 'REQUEST_SESSION_NOT_FOUND',
+					result: { user_details: processedUserDetails },
+				})
+			}
+
+			requestSessions.user_details = processedUserDetails
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'REQUEST_SESSION_DETAILS',
+				result: requestSessions,
+			})
+		} catch (error) {
+			console.error(error)
+			throw error
+		}
+	}
+
+	static async userAvailability(userId, page, limit, search, status, roles) {
```

Comment:
@sumanvpacewisdom I should get my availability on start and end date 

sample req 
http://localhost:7101/api/v1/mentor/availability/123&startDate=2024-01-12&endDate=2024-02-02